xactions.py

```diff
@@ -2057,6 +2057,17 @@ def run(self, count=1, mode=None):
         self.window.run_command('tab_control', {'command': 'prev'})
         self.window.run_command('_enter_normal_mode', {'mode': mode})
 
+# TODO <C-]> should learn visual mode
+# TODO <C-]> should learn to count
+class _vi_ctrl_right_square_bracket(ViWindowCommandBase):
+    """
+    http://vimdoc.sourceforge.net/htmldoc/tagsrch.html#CTRL-]
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def run(self):
+        self.window.run_command('goto_definition')
```

Comment:
To be honest there are few very similar goto/jump tag commands in vim, some of which I can't tell the difference. 

As far as I understand `gd` is local where `ctrl+]` will jump to declaration in other files. `gd` is also much smarter:

> gd                      Goto local Declaration.  When the cursor is on a local
                        variable, this command will jump to its declaration.
                        First Vim searches for the start of the current
                        function, just like "[[".  If it is not found the
                        search stops in line 1.  If it is found, Vim goes back                                                              
                        until a blank line is found.  From this position Vim
                        searches for the keyword under the cursor, like with
                        "*", but lines that look like a comment are ignored